src/Linux/Core/Impl/IO/FileSystem.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Common.Core.IO;
+using System;
+using Microsoft.Extensions.FileSystemGlobbing;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.IO.Compression;
+using Microsoft.Common.Core;
+using System.Diagnostics;
+
+namespace Microsoft.R.Common.Core.Linux {
+    public class FileSystem : IFileSystem {
+
+        public IFileSystemWatcher CreateFileSystemWatcher(string path, string filter) => new FileSystemWatcherProxy(path, filter);
+
+        public IDirectoryInfo GetDirectoryInfo(string directoryPath) => new DirectoryInfoProxy(directoryPath);
+
+        public bool FileExists(string path) => File.Exists(path);
+
```

Comment:
Looks like mostly same thing as Windows. Perhaps just have common in Core or at least a base class if implementations are slightly different?